ch2driver/pytpcc/drivers/nestcollectionsdriver.py

```diff
@@ -714,79 +935,122 @@ def loadTuples(self, tableName, tuples):
         return
 
     def getOneDoc(self, tableName, tuple, denorm):
-         columns = TABLE_COLUMNS[tableName]
+        if self.schema == constants.CH2_DRIVER_SCHEMA["CH2"]:
+            return self.getOneCH2Doc(tableName, tuple, denorm)
+        else:
+            return self.getOneCH2PPDoc(tableName, tuple, denorm)
+
+    def getOneCH2Doc(self, tableName, tuple, denorm):
+        columns = TABLE_COLUMNS[tableName]
+        key = ""
+        if denorm:
+            for l, k in enumerate(KEYNAMES[tableName]):
+                if l == 0:
+                    key = str(tuple[columns[k]])
+                else:
+                    key = key + '.' + str(tuple[columns[k]])
+            val = tuple
+        else:
+            for l, k in enumerate(KEYNAMES[tableName]):
+                if l == 0:
+                    key = str(tuple[k])
+                else:
+                    key = key + '.' + str(tuple[k])
+            val = {}
+            for l, v in enumerate(tuple):
+                v1 = tuple[l]
+                if tableName == constants.TABLENAME_ORDERS and columns[l] == "o_orderline":
+                    v1 = []
+                    for olv in v:
+                        v1.append(self.genNestedTuple(olv, constants.TABLENAME_ORDERLINE))
+
+                elif isinstance(v1,(datetime)):
+                    v1 = str(v1)
+                val[columns[l]] = v1
+
+        return key, val
+
+    def getOneCH2PPDoc(self, tableName, tuple, denorm):
+        columns = CH2PP_TABLE_COLUMNS[tableName]
+        key = ""
+        if denorm:
+            for l, k in enumerate(KEYNAMES[tableName]):
+                if l == 0:
+                    key = str(tuple[columns[k]])
+                else:
+                    key = key + '.' + str(tuple[columns[k]])
+            val = tuple
+        else:
+            for l, k in enumerate(KEYNAMES[tableName]):
+                if l == 0:
+                    key = str(tuple[k])
+                else:
+                    key = key + '.' + str(tuple[k])
+            val = {}
+            for l, v in enumerate(tuple):
+                v1 = tuple[l]
+                if isinstance(v1,(datetime)):
+                    v1 = str(v1)
+                elif tableName == constants.TABLENAME_ORDERS and columns[l] == "o_orderline":
+                    v1 = []
+                    for olv in v:
+                        v1.append(self.genNestedTuple(olv, constants.TABLENAME_ORDERLINE))
+                elif (self.schema == constants.CH2_DRIVER_SCHEMA["CH2P"] or
+                      self.schema == constants.CH2_DRIVER_SCHEMA["CH2PP"]):
```

Comment:
This shouldn't be necessary as this whole method is only called if this condition is true

Reply:
Will do this in a separate merge.